Newsletters: add 292 (2024-03-06) (#1537)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2024-03-06-newsletter.md

  The specification requires the body immediately after the colon to be
  a legacy P2PKH Bitcoin address, e.g. `bitcoin:1BoB...`.  After the
  body, additional parameters can be passed using HTTP query encoding,
  including addresses for non-legacy address formats.  For example, a
  bech32m address could be `bitcoin:1Bob...?bech32m=bc1pbob...`.
  However, this is significantly different from how `bitcoin:` URIs are
  being used.  Non-P2PKH addresses are often used as the body and
  sometimes the body is left blank by software that only wants to
  receive payment through an alternative payment protocol.
  Additionally, Baker notes that `bitcoin:` URIs are increasingly being
  used to transmit privacy-respecting persistent identifiers, such as
  those for [silent payments][topic silent payments] and [offers][topic
  offers].

    As discussed on the thread, an improvement could be having the
    creator of the URI specify all of the payment methods they supported
    using bare parameters, for example: `bitcoin:?bc1q...&sp1q...`.  The
    spender (who is usually responsible for paying fees) could then
    choose their preferred payment method from the list.  Although some

>     minor technical points were still being discussed at the time

Reviewer: ```suggestion
    minor technical points were still being discussed at the time of
```

    writing, no major criticisms of this approach had been posted.

- **PSBTs for multiple concurrent MuSig2 signing sessions:** Salvatore
  Ingala [posted][ingala musig2] to Delving Bitcoin about minimizing the
  amount of state needed to perform multiple [MuSig2][topic musig]
  signing sessions in parallel.  Using the signing algorithm described
  in [BIP327][], a group of co-signers will need to temporarily store a
  linearly increasing amount of data for each additional input they want
  to add to a transaction they create.  With many hardware signing
  devices limited in the amount of storage they have available, it would
  be very useful to minimize the amount of state required (without
  reducing security).

  Ingala proposes generating a single state object for an entire
  [PSBT][topic psbt] and then deterministically deriving the per-input
  state from it in a way that still makes the result indistinguishable
  from random.  That way the amount of state a signer needs to store is

>   constant no matter how inputs a transaction has.

Reviewer: ```suggestion
  constant no matter how many inputs a transaction has.
```

  In a [reply][scott musig2], developer Christopher Scott noted that
  [BitEscrow][] already uses a similar mechanism.

- **Discussion about adding more BIP editors:** Ava Chow [posted][chow
  bips] to the Bitcoin-Dev mailing list to suggest the addition of BIP
  editors to help the current editor.  The current editor, Luke Dashjr,
  [says][dashjr backlogged] he is backlogged and has requested help.
  Chow suggested two well-known expert contributors to become editors,
  which seemed to have support.  Also discussed was whether the
  additional editors should have the ability to assign BIP numbers.  No
  clear resolution had been reached as of this writing.

- **GitLab backup for Bitcoin Core GitHub project:** Fabian Jahr
  [posted][jahr gitlab] to Delving Bitcoin about maintaining a backup of
  the Bitcoin Core project's GitHub account on an self-hosted GitLab
  instance.  In case the project ever needed to leave GitHub suddenly,
  this would make all existing issues and pull requests accessible on
  GitLab within a short amount of time, allowing work to continue with
  only a brief interruption.  Jahr provided a preview of the project on

>   GitLab and plans to keep backups going forward to allow rapidly

Reviewer: Seems fine as is.

release candidates.*

- Eclair v0.10.0 ? FIXME:harding

<!-- FIXME:harding to update Tuesday -->

## Notable code and documentation changes

_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Bitcoin Inquisition][bitcoin inquisition repo], and [BINANAs][binana
repo]._

- [Bitcoin Core #29412][] adds code that checks for every known way to
  take a valid block, mutate it, and produce an alternative block that
  is invalid but has the same block header hash.  Mutated blocks have
  caused multiple vulnerabilities in the past.  In 2012, and again in
  2017, Bitcoin Core's cached rejection of invalid blocks meant that an
  attacker could take a new valid block, mutate it into an invalid
  block, and submit it to a victim's node; that node would reject it as
  invalid and would not later accept the valid form of the block (until

>   the node was next restarted), forking it off the best block chain and

Author: Leaving as-is for this newsletter.  I left a comment on #1541 and, if there's no major disagreement there, I'll switch to "blockchain" going forward.

  additional editors should have the ability to assign BIP numbers.  No
  clear resolution has been reached as of this writing.

- **GitLab backup for Bitcoin Core GitHub project:** Fabian Jahr
  [posted][jahr gitlab] to Delving Bitcoin about maintaining a backup of
  the Bitcoin Core project's GitHub account on a self-hosted GitLab
  instance.  In case the project ever needed to leave GitHub suddenly,
  this would make all existing issues and pull requests accessible on
  GitLab within a short amount of time, allowing work to continue with
  only a brief interruption.  Jahr provided a preview of the project on
  GitLab and plans to keep backups going forward to allow rapidly
  switching to GitLab if necessary.  His post has not received any
  comments as of this writing, but we thank him for making a potential
  transition as easy as possible.

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [Eclair v0.10.0][] is a new major release of this LN node
  implementation.  It "adds official support for the [dual-funding

>   feature][topic dual funding], an up-to-date implementation of Bolt 12

Reviewer: ```suggestion
  feature][topic dual funding], an up-to-date implementation of BOLT12
```
